src/SnD.Sdk/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,6 +503,101 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
+        /// <returns> A V1PodFailurePolicyRule object.</returns>
+        public static V1PodFailurePolicyRule ConvertToFailurePolicyRule(KeyValuePair<string, List<int>> action)
+        {
+            return new V1PodFailurePolicyRule(
+                action: action.Key,
+                onExitCodes: new V1PodFailurePolicyOnExitCodesRequirement(operatorProperty: "In",
+                    values: action.Value));
+        }
+
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actions">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job, Dictionary<string, List<int>> actions)
+        {
+            job.Spec ??= new V1JobSpec();
+            job.Spec.PodFailurePolicy ??= new V1PodFailurePolicy();
+
+            job.Spec.PodFailurePolicy.Rules = (job.Spec.PodFailurePolicy.Rules ?? new List<V1PodFailurePolicyRule>())
+                .Concat(actions.Select(ConvertToFailurePolicyRule))
+                .GroupBy(rule => rule.Action)
+                .Select(grp => grp.Aggregate(MergePodFailurePolicyRules))
+                .ToList();
+
+            return job;
+        }
+
+        /// <summary>
+        /// Attaches a Disruption Target failure to a given list of action.
+        /// </summary>
+        /// <param name="action">A key-value pair with the action name and a list of exit codes to attach a disruption target.</param>
+        /// <returns> A V1PodFailurePolicyRule object.</returns>
+        public static V1PodFailurePolicyRule AttachDisruptionTargetFailure(KeyValuePair<string, List<int>> action)
+        {
+            return new V1PodFailurePolicyRule(
+                action: action.Key,
+                onExitCodes: new V1PodFailurePolicyOnExitCodesRequirement(operatorProperty: "In", values: action.Value),
+                onPodConditions: new List<V1PodFailurePolicyOnPodConditionsPattern>
+                {
+                    new(status: "True", type: "DisruptionTarget")
+                });
+        }
+
+        /// <summary>
+        /// Attaches disruption target to target pod policy failure actions and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actions">A list with all action names and corresponding exit code number to attach disruption target.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules with disruption target.</returns>>
+        public static V1Job WithPodPolicyFailureDisruptionTargetFailure(this V1Job job, Dictionary<string, List<int>> actions)
```

Comment:
This is wrong as well

Reply:
I've refactored it in [561f1cf](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/561f1cf4a699fe803d15ae5b14f721b523caee0e)
I hope it is doing its job now